connectors/src/connectors/webcrawler/index.ts

```diff
@@ -32,6 +37,10 @@ export async function createWebcrawlerConnector(
   dataSourceConfig: DataSourceConfig,
   urlConfig: CreateConnectorUrlRequestBody
 ): Promise<Result<string, Error>> {
+  const depth = urlConfig.depth;
+  if (!isDepthOption(depth)) {
```

Comment:
Let's move this logic to the io-ts codec. I added a range codec, like this:

https://github.com/dust-tt/dust/blob/fae4be7564994c04dd77dc019c60aba3e841480a/types/src/front/api_handlers/internal/agent_configuration.ts#L7